WINC 1176: Add Windows node limitations (#49152)
openshift/openshift-docs: modules/windows-containers-release-notes-limitations.adoc

// Module included in the following assemblies:
//
// * windows_containers/windows-containers-release-notes-#-x

[id="windows-containers-release-notes-limitations"]
= Known limitations

Note the following limitations when working with Windows Container Support for Red Hat OpenShift:

* The following {product-title} features are not supported:
** OpenShift Serverless
** Red Hat OpenShift Pipelines
** Thanos User Workload Monitoring
** Service Mesh
** Cost Management
** CodeReady (Containers and Workspaces)
** Odo
** Builds v2 or BuildConfig or s2i

> ** Any operating System Red Hat ships

Reviewer: Not sure what this bullet is about.

* Windows Nodes do not support workloads created by using deployment configs. You can use a deployment or other method to deploy workloads.

* The WMCO is not supported in clusters that use a cluster-wide proxy because the WMCO is not able to route traffic through the proxy connection for the workloads.

* Kubernetes has identified the following link:https://kubernetes.io/docs/concepts/windows/intro/#limitations[node feature limitations] :
** HugePages: not supported for Windows containers
** Privileged containers: not supported for Windows containers. HostProcess Containers offer similar functionality.

> ** TerminationGracePeriod: requires containerD

Reviewer: We can remove this bullet given we are using containerd.

Author: @aravindhp Does this list apply to all versions of WMCO? I was thinking we might need this for versions earlier than 6.0.0??

Reviewer: That is correct. BTW, it should be `containerd`. Upstream needs to be fixed.

// Module included in the following assemblies:
//
// * windows_containers/windows-containers-release-notes-#-x

[id="windows-containers-release-notes-limitations_{context}"]
= Known limitations

Note the following limitations when working with Windows nodes managed by the WMCO (Windows nodes):

* The following {product-title} features are not supported on Windows nodes:
** Red Hat OpenShift Developer CLI (odo)
** Image builds
** OpenShift Pipelines
** OpenShift Service Mesh
** OpenShift monitoring of user-defined projects

* The following Red Hat features are not supported on Windows nodes:
** link:https://access.redhat.com/documentation/en-us/cost_management_service/2022/html/getting_started_with_cost_management/assembly-introduction-cost-management?extIdCarryOver=true&sc_cid=701f2000001OH74AAG#about-cost-management_getting-started[Red Hat cost management]
** link:https://developers.redhat.com/products/openshift-local/overview[Red Hat OpenShift Local]

> * Windows nodes do not support pulling container images from private registries. It is recommended that you use images from public registries or pre-pull the images.

Reviewer 2: ```suggestion
* Windows nodes do not support pulling container images from private registries. You can use images from public registries or pre-pull the images.
```
Maybe can be worded like this to avoid the "recommended". This wording is similar what you have in the following bullet. Up to you though!

Author: @kelbrown20 Good idea. Believe it or not, _it is recommended_ is the recommended phrase for us to use.

Reviewer 2: Yeah that was always confusing for me when I'm going through the guides 😄
